cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -18,7 +18,7 @@ set(CMAKE_SYSTEM_NAME Emscripten)
 set(CMAKE_SYSTEM_VERSION 1)
 
 set(CMAKE_CROSSCOMPILING TRUE)
-set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS FALSE)
+set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
```

Comment:
I'm not quite sure that the effect of this will be... is there some way we can have folks opt into this rather than making it the default.

There is some discussion about weather we want to more broadly declare the SIDE_MODULE support is good enough for general usage:  https://github.com/emscripten-core/emscripten/issues/14408.